library/src/methods/getFallbacks/getFallbacks.ts

```diff
@@ -64,6 +64,12 @@ export function getFallbacks<
         ErrorMessage<TupleWithRestIssue> | undefined
       >,
 >(schema: TSchema): InferFallbacks<TSchema> {
+  
+  // If it's and object schema and it has a fallback, return it
+  if (schema.type === 'object' && 'fallback' in schema) {
+    return schema.fallback as InferFallbacks<TSchema>;
+  }
+
   // If it is an object schema, return fallbacks of entries
   if ('entries' in schema) {
     const object: Record<string, unknown> = {};
```

Comment:
Ideally, it would be like this, however `fallback` requires all values to be defined. 

```suggestion
  // If it is an object schema, return fallbacks of entries
  // If an object itself has a fallback, use it
  if ('entries' in schema) {
    const object: Record<string, unknown> = schema.type === 'object' && 'fallback' in schema ? schema.fallback : {};
```

eg.

```ts
const ObjectWithFallbackSchema = fallback(object({
  foo: number(),
  bar: fallback(number(), 5)
}), {
  foo: 3
});
```

I would expect `bar` to not be required, because it itself has a fallback.

Reply:
It can be achieve by adjusting `Fallback` type to

```ts
/**
 * Fallback type.
 */
export type Fallback<
  TSchema extends BaseSchema<unknown, unknown, BaseIssue<unknown>>,
  TFallbackSchema = MaybeReadonly<InferOutput<TSchema>>
> =
  | {
    [key in keyof TFallbackSchema as TFallbackSchema[key] extends SchemaWithFallback<any, any> ? never : key]: TFallbackSchema[key];
    } & {
      [key in keyof TFallbackSchema as TFallbackSchema[key] extends SchemaWithFallback<any, any> ? key : never]?: TFallbackSchema[key];
    }
  | ((
      dataset?: OutputDataset<InferOutput<TSchema>, InferIssue<TSchema>>,
      config?: Config<InferIssue<TSchema>>
    ) => ({
      [key in keyof TFallbackSchema]: TFallbackSchema[key] extends SchemaWithFallback<any, any>
      ? TFallbackSchema[key]
      : TFallbackSchema[key];
    } & {
      [key in keyof TFallbackSchema as TFallbackSchema[key] extends SchemaWithFallback<any, any> ? key : never]?: TFallbackSchema[key];
    }));
```

but that's whole other discussion